Assets/Plugins/CountlySDK/Services/ConsentCountlyService.cs

```diff
@@ -1,47 +1,310 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
 using Plugins.CountlySDK.Enums;
-using Plugins.CountlySDK.Models;
+using Plugins.CountlySDK.Models;
+using UnityEditor;
+using UnityEngine;
+
+namespace Plugins.CountlySDK.Services
+{
+    public class ConsentCountlyService : AbstractBaseService
+    {
+        internal bool RequiresConsent { get; private set; }
+        private readonly CountlyConfiguration _config;
+        private readonly Dictionary<Features, bool> _modifiedConsents;
+        private readonly Dictionary<Features, bool> _countlyFeatureConsents;
+        private readonly Dictionary<string, Features[]> _countlyFeatureGroups;
+
+        internal ConsentCountlyService(CountlyConfiguration config, ConsentCountlyService consentService) : base(consentService)
+        {
+            _modifiedConsents = new Dictionary<Features, bool>();
+            _countlyFeatureConsents = new Dictionary<Features, bool>();
+            _countlyFeatureGroups = new Dictionary<string, Features[]>();
+
+            _config = config;
+
+            RequiresConsent = _config.RequiresConsent;
+            _countlyFeatureGroups = new Dictionary<string, Features[]>(_config.FeatureGroups);
+
+            GiveConsentInternal(_config.Features);
 
-namespace Plugins.CountlySDK.Services
-{
-    public class ConsentCountlyService : IBaseService
-    {
-
-        internal ConsentCountlyService()
-        { }
-
-        #region Consents
-
-        #region Unused Code
-
-        /// <summary>
-        /// Checks consent for a particular feature
-        /// </summary>
-        /// <param name="feature"></param>
-        /// <returns></returns>
-        public bool CheckConsent(Features feature)
+        }
+
+        #region Public Methods
+
+        /// <summary>
+        /// Checks consent for a particular feature
+        /// </summary>
+        /// <param name="feature">a feature for which consent should be checked</param>
+        /// <returns>bool</returns>
+        public bool CheckConsent(Features feature)
+        {
+            return !RequiresConsent || (_countlyFeatureConsents.ContainsKey(feature) && _countlyFeatureConsents[feature]);
+        }
+
+        internal bool AnyConsentGiven()
         {
-            return true;// ConsentModel.CheckConsent(feature.toString());
-        }
+            if (!RequiresConsent) {
+                //no consent required - all consent given
+                return true;
+            }
 
-        public void DeviceIdChanged(string deviceId, bool merged)
-        {
-            
-        }
+            foreach (KeyValuePair<Features, bool> entry in _countlyFeatureConsents) {

```

Comment:
"_countlyFeatureConsents" is a dictionary, use it like one. 
Don't go over every value like for an array.

Reply:
yah, i know it is a dictionary, but we checkping if any Consent is enabled not a specific one.

Reply:
ok